src/main.rs

```diff
@@ -49,28 +52,42 @@ use libtock_drivers::usb_ctap_hid;
 
 libtock_core::stack_size! {0x4000}
 
-const KEEPALIVE_DELAY_MS: isize = 100;
-const KEEPALIVE_DELAY: Duration<isize> = Duration::from_ms(KEEPALIVE_DELAY_MS);
+const KEEPALIVE_DELAY_MS: ClockInt = 100;
+const KEEPALIVE_DELAY: Milliseconds<ClockInt> = Milliseconds(KEEPALIVE_DELAY_MS);
+const KEEPALIVE_DELAY_TOCK: Duration<isize> = Duration::from_ms(KEEPALIVE_DELAY_MS as isize);
 const SEND_TIMEOUT: Duration<isize> = Duration::from_ms(1000);
 
+pub struct LibtockClock<'a, const CLOCK_FREQUENCY: u32>(libtock_drivers::timer::Timer<'a>);
+impl<const CLOCK_FREQUENCY: u32> embedded_time::Clock for LibtockClock<'_, CLOCK_FREQUENCY> {
+    type T = ClockInt;
+    const SCALING_FACTOR: embedded_time::fraction::Fraction =
+        <embedded_time::fraction::Fraction>::new(1, CLOCK_FREQUENCY);
+    fn try_now(&self) -> Result<embedded_time::Instant<Self>, embedded_time::clock::Error> {
+        let timer = &self.0;
+        let now = timer.get_current_clock().flex_unwrap();
+        Ok(embedded_time::Instant::new(now.num_ticks() as Self::T))
+    }
+}
+
 fn main() {
     // Setup the timer with a dummy callback (we only care about reading the current time, but the
     // API forces us to set an alarm callback too).
     let mut with_callback = timer::with_callback(|_, _| {});
     let timer = with_callback.init().flex_unwrap();
+    let clock = LibtockClock::<32768>(timer);
```

Comment:
This is unfortunately not possible because the trait `embedded_time::Clock` cannot be made into an object, because it contains this associated `const`: `const SCALING_FACTOR: Fraction;`.

Reply:
Oh I wasn't aware of this limitation. The clock module should probably look like:

```rust
// Should implement embedded_time::Clock
pub type Clock = ...
pub fn new_clock() -> Clock { ... }
```

Reply:
We cannot name "our" clock `Clock` because this would lead to a conflict with the trait name `Clock`  from `embedded_time`, which is used by ` try_now`. So currently I re-export that trait in the module `clock` and name the clock implementation `CtapClock`. We could use `Clock` for this one and renamed the trait on re-export but that seems more cumbersome than necessary and prone to confusion for existing users of `embedded_time`.
Regarding the `new_clock` function, I added it. Note that it must take a timer with its lifetime as parameter, when used without `std`.

Reply:
Yes, `Clock` was just an example. `CtapClock` is good.